micro/core/params.py

```diff
@@ -39,34 +93,66 @@ def __priority_param(self, cli_param, env_var_name, config_key):
         if self.__config.key(config_key):
             return self.__config.key(config_key)
 
-        return self.__default[config_key]
-
-    def broker_url(self):
-        return self.__priority_param(self.__args.broker_url,
-                                     "MICRO_BROKER_URL",
-                                     "broker_url")
-
-    def queue_name(self):
-        return self.__priority_param(self.__args.queue_name,
-                                     "MICRO_QUEUE_NAME",
-                                     "queue_name")
-
-    def hostname(self):
-        return self.__priority_param(self.__args.hostname,
-                                     "MICRO_HOSTNAME",
-                                     "hostname")
-
-    def num_workers(self):
-        return int(self.__priority_param(self.__args.num_workers,
-                                         "MICRO_NUM_WORKERS",
-                                         "num_workers"))
-
-    def log_path(self):
-        return self.__priority_param(self.__args.log_path,
-                                     "MICRO_CELERY_LOG_PATH",
-                                     "log_path")
-
-    def pid_path(self):
-        return self.__priority_param(self.__args.pid_path,
-                                     "MICRO_CELERY_PID_PATH",
-                                     "pid_path")
+        return DEFAULT[config_key]
+
+    def __set_all(self):
+        for p in PARAMS:
+            value = self.__priority_param(self.__args.get(PARAMS[p]["var"]),
+                                          PARAMS[p]["env"],
+                                          PARAMS[p]["var"])
+            if value:
+                envvar = "_" + PARAMS[p]["env"]
+                os.environ[envvar] = str(value)
+
+    def __check_required(self):
```

Comment:
Here the previous comment is valid too ;)

Reply:
solved in commit b2f08aba5dff0c10a7c217c824e0086e5472c060